src/osip-monitoring-controller/src/main/java/edu/kit/pse/osip/monitoring/controller/MonitoringController.java

```diff
@@ -80,14 +121,442 @@ private void setupUI(Stage stage) {
         AboutDialog aboutDialog = new AboutDialog();
         currentView.setMenuAboutButtonHandler(event -> aboutDialog.show());
         currentSettingsView.setSettingsCancelButtonHandler(event -> currentSettingsView.hideSettingsWindow());
-//        currentSettingsView.setSettingsSaveButtonHandler(new SettingsSaveButtonHandler(currentSettings,
-//                currentView, currentSettingsView));
+        currentSettingsView.setSettingsSaveButtonHandler(event -> handleSaveButton(currentSettings,
+                currentView, currentSettingsView));
     }
 
     /**
      * Called when the last window is closed.
      */
     public void stop() {
-        throw new RuntimeException("Not implemented!");
+        System.out.println("Stopped monitoring thread");
+        for (TankContainer cont : tanks) {
+            try {
+                cont.client.disconnectClient();
+            } catch (UAClientException e) {
+                System.err.println(e.getMessage());
```

Comment:
Ich denke Exceptions sagen oft, was genau schief geht, aber nicht in welchem Kontext es passiert. Deswegen bin ich dafür die Ausgabe immer so zu machen: 
`System.err.println(Error while disconnecting: " + e.getMessage());`
(Das hab ich weiter oben ja auch schon mal bemängelt)